roles/container-engine/docker/tasks/main.yml

```diff
@@ -235,6 +235,14 @@
         resolvconf_mode == 'docker_dns' and
         installed_docker_version.stdout is version('1.12', '<')
 
+- name: Install Docker plugin
+  command: docker plugin install --grant-all-permissions {{ docker_plugin | quote }}
+  when: docker_plugin is defined
+  register: docker_plugin_status
+  failed_when:
+    - docker_plugin_status.failed 
+    - '"already exists" not in docker_plugin_status.stderr'
```

Comment:
if plugin is installed it really needs to fail?

Reply:
@Atoms No, this achieves the opposite. It is only considered as failure if there is an stderr message different from "already exists".  This way it runs idempotently.

Reply:
ah missed 'not in' :)